src/software/ai/hl/stp/play/play.cpp

```diff
@@ -139,6 +143,41 @@ std::unique_ptr<TbotsProto::PrimitiveSet> Play::get(
         }
     }
 
+    /* substitute injured robots when the game is in play */
+    if (injured_robots.size() > 0)
+    {
+        unsigned int num_injured = (unsigned int)injured_robots.size();
```

Comment:
we don't want to substitute all the injured robots at once, I think we should substitute robots one-at-a-time. Let's define some dynamic params in `proto/parameters.proto`:

```
max_simultaneous_robot_substitution
enable_robot_substitution
```

and use those dynamic params in the substitution logic

Reply:
bump @jameschuang2002